win32/tests/job_object_helper_int/job_object_helper_int.c

```diff
@@ -92,6 +99,235 @@ TEST_FUNCTION_CLEANUP(cleanup)
+    ASSERT_ARE_NOT_EQUAL(int, length, 0, "GetModuleFileNameA failed");
+    /* Get the full path to the executable// Copy the full path to directory and remove the executable name */
+    strcpy(directory, path);
+    for (int i = length - 1; i >= 0; --i)
+    {
+        if (directory[i] == '\\')
+        {
+            directory[i] = '\0';
+            break;
+        }
+    }
+
+    // Create the full path to the job_object_helper_tester executable
+    char fullPath[MAX_PATH];
+    (void)snprintf(fullPath, sizeof(fullPath), "%s\\..\\job_object_helper_tester\\job_object_helper_tester.exe", directory);
+
+    // start 3 new process using CreateNewProcess, those processes shall
+    // call job_object_helper_set_job_limits_to_current_process
+    STARTUPINFOA si[NUM_TEST_PROCESSES];
+    PROCESS_INFORMATION pi[NUM_TEST_PROCESSES];
+
+    for (int i = 0; i < NUM_TEST_PROCESSES; ++i)
+    {
+        LogInfo("Starting process %d", i);
+        ZeroMemory(&si[i], sizeof(si[i]));
+        si[i].cb = sizeof(si[i]);
+        ZeroMemory(&pi[i], sizeof(pi[i]));
+
+        char cmdLine[512];
+        (void)snprintf(cmdLine, sizeof(cmdLine), "\"%s\" %s %d %d", fullPath, job_name, 50, 50);
+
+        /* Start the process; this process calls job_object_helper_set_job_limits_to_current_process */
+        (void)CreateProcessA(NULL, cmdLine, NULL, NULL, FALSE, DETACHED_PROCESS, NULL, NULL, &si[i], &pi[i]);
+
+        /* sleep for 2 seconds to allow the child process to run */
+        Sleep(2000);
+        /* try for 3 times before declaring a failure, as it may take time for a new process to get scheduled and create the job object  */
+        HANDLE job_object = NULL;
+        for (int j = 0; j < NUM_RETRY; ++j)
+        {
+            job_object = OpenJobObjectA(JOB_OBJECT_QUERY, FALSE, job_name);
+            if (job_object != NULL)
+            {
+                break;
+            }
+            LogInfo("OpenJobObjectA failed, will retry %d", j + 1);
+            /* sleep for 2 seconds */
+            Sleep(2000);
+        }
+
+        ASSERT_IS_NOT_NULL(job_object);
+        ASSERT_IS_TRUE(job_object != NULL, "Failed to open job object");
+
+        /* Query the Job Object to check if it has all processes associated with it */
+        DWORD bufferSize = sizeof(JOBOBJECT_BASIC_PROCESS_ID_LIST) + sizeof(ULONG_PTR) * 16;
+        JOBOBJECT_BASIC_PROCESS_ID_LIST* pidList = NULL;
+        while (1)
+        {
+            pidList = (JOBOBJECT_BASIC_PROCESS_ID_LIST*)malloc(bufferSize);
+
+            if (!QueryInformationJobObject(job_object, JobObjectBasicProcessIdList, pidList, bufferSize, NULL))
+            {
+                DWORD err = GetLastError();
+                if (err == ERROR_MORE_DATA) {
+                    // Increase buffer size and try again
+                    bufferSize *= 2;
+                    free(pidList);
+                    continue;
+                }
+            }
+            break;
+        }
```

Comment:
I suspect `NumberOfAssignedProcesses` field of `JOBOBJECT_BASIC_PROCESS_ID_LIST` will tell you exactly how many processes are assigned to the Job. You could use a static variable and look the other way about the other fields of `JOBOBJECT_BASIC_PROCESS_ID_LIST`